pkg/render/amazoncloudintegration.go

```diff
@@ -0,0 +1,313 @@
+	return &corev1.ServiceAccount{
+		TypeMeta: metav1.TypeMeta{Kind: "ServiceAccount", APIVersion: "v1"},
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      AmazonCloudIntegrationComponentName,
+			Namespace: AmazonCloudIntegrationNamespace,
+		},
+	}
+}
+
+// clusterRole creates a clusterrole that gives permissions to access backing CRDs and
+// k8s networkpolicies.
+func (c *amazonCloudIntegrationComponent) clusterRole() *rbacv1.ClusterRole {
+	return &rbacv1.ClusterRole{
+		TypeMeta: metav1.TypeMeta{Kind: "ClusterRole", APIVersion: "rbac.authorization.k8s.io/v1"},
+		ObjectMeta: metav1.ObjectMeta{
+			Name: AmazonCloudIntegrationComponentName,
+		},
+		Rules: []rbacv1.PolicyRule{
+			{
+				APIGroups: []string{"*"},
+				Resources: []string{
+					"nodes",
+				},
+				Verbs: []string{
+					"get",
+					"list",
+					"watch",
+				},
+			},
+			{
+				APIGroups: []string{"crd.projectcalico.org"},
+				Resources: []string{
+					"globalnetworkpolicies",
+					"networkpolicies",
+					"tiers",
+					"clusterinformations",
+					"hostendpoints",
+					"licensekeys",
+				},
+				Verbs: []string{
+					"create",
+					"get",
+					"list",
+					"update",
+					"watch",
+					"delete",
+				},
+			},
+		},
+	}
+}
+
+// clusterRoleBinding creates a clusterrolebinding that applies apiServiceAccountClusterRole to
+// the tigera-apiserver service account.
+func (c *amazonCloudIntegrationComponent) clusterRoleBinding() *rbacv1.ClusterRoleBinding {
+	return &rbacv1.ClusterRoleBinding{
+		TypeMeta: metav1.TypeMeta{Kind: "ClusterRoleBinding", APIVersion: "rbac.authorization.k8s.io/v1"},
+		ObjectMeta: metav1.ObjectMeta{
+			Name: "tigera-apiserver-access-crds",
+		},
+		Subjects: []rbacv1.Subject{
+			{
+				Kind:      "ServiceAccount",
+				Name:      AmazonCloudIntegrationComponentName,
+				Namespace: AmazonCloudIntegrationNamespace,
+			},
+		},
+		RoleRef: rbacv1.RoleRef{
+			Kind:     "ClusterRole",
+			Name:     AmazonCloudIntegrationComponentName,
+			APIGroup: "rbac.authorization.k8s.io",
+		},
+	}
+}
+
+func (c *amazonCloudIntegrationComponent) credentialSecret() *corev1.Secret {
+	if c.credentials == nil {
+		return nil
+	}
+	return &corev1.Secret{
+		TypeMeta: metav1.TypeMeta{Kind: "Secret", APIVersion: "v1"},
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      AmazonCloudIntegrationCredentialName,
+			Namespace: AmazonCloudIntegrationNamespace,
+		},
+		Data: map[string][]byte{
+			AmazonCloudCredentialKeyIdName:     c.credentials.KeyId,
+			AmazonCloudCredentialKeySecretName: c.credentials.KeySecret,
+		},
+	}
+}
+
+// deployment creates a deployment containing the API and query servers.
+func (c *amazonCloudIntegrationComponent) deployment() *appsv1.Deployment {
+	var replicas int32 = 1
+
+	d := &appsv1.Deployment{
+		TypeMeta: metav1.TypeMeta{Kind: "Deployment", APIVersion: "v1"},
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      AmazonCloudIntegrationComponentName,
+			Namespace: APIServerNamespace,
```

Comment:
This would have been a fun mistake
```suggestion
			Namespace: AmazonCloudIntegrationNamespace,
```